pkg/auth/spire/certificate_provider_test.go

```diff
@@ -389,8 +390,24 @@ func TestSpireDelegateClient_GetCertificateForIdentity(t *testing.T) {
 				t.Errorf("SpireDelegateClient.GetCertificateForIdentity() error = %v, wantErr %v", err, tt.wantErr)
 				return
 			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("SpireDelegateClient.GetCertificateForIdentity() = %v, want %v", got, tt.want)
+			if tt.want != nil {
+				if !assert.Equal(t, tt.want.Certificate, got.Certificate) {
+					t.Errorf("SpireDelegateClient.GetCertificateForIdentity().Certificate = %v, want %v", got.Certificate, tt.want.Certificate)
+				}
+				wantKey := tt.want.PrivateKey.(*rsa.PrivateKey)
+				gotKey := got.PrivateKey.(*rsa.PrivateKey)
+				if !wantKey.Equal(gotKey) {
+					if !assert.Equal(t, tt.want.PrivateKey, got.PrivateKey) {
```

Comment:
you can add a message to the assert functions instead of checking their result and emitting an additional error.

```
assert.Equal(t, tt.want.PrivateKey, got.PrivateKey, "SpireDelegateClient.GetCertificateForIdentity().PrivateKey = %v, want %v", got.PrivateKey, tt.want.PrivateKey)
```

Reply:
Yeah that can be helpful in general. For many of these cases, `assert.Equal` will already emit the relevant information for us in a standardized form. Where this PR uses `assert.Equal`, I'll just drop the message entirely.